lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -2610,3 +2611,457 @@ SILFunction *swift::lookupPrespecializedSymbol(SILModule &M,
   return Specialization;
 }
 
+/// Create a new function name for the newly generated protocol constrained generic function.
+std::string ExistentialSpecializerTransform::createExistentialSpecializedFunctionName(){
```

Comment:
OK. So, You are recommending creating a new Mangler for ExistentialSpecializer?

Reply:
Look at how function signature optimizations mangle which parameters were exploded, or were dead, etc. You need to do the same.

Reply:
I see. It looks to me that I can reuse the Mangle::FunctionSignatureSpecializationMangler. Did you think otherwise that I create a separate Mangler altogether?

Reply:
Are there any tutorials/docs around adding a new a new ArgumentModifier in the Mangler? I am running into "Can't demangle" issues.